### PR TITLE
fix(tsoa): server port keeps changing between compilations

### PR DESCRIPTION
**Pull request by eladcon**
saving the last used port in a bucket so it could be used for later compilations

### CODE REVIEW

**Comment by ekeren** [comment]
Making sure you have tests as a use case on our plate

File: tsoa/sim.w [hunk 1]
```diff
@@ -27,6 +28,7 @@ pub class Service_sim impl types.IService {
     }
 
     this.state = new sim.State();
+    this.bucket = new cloud.Bucket();
```
Comment (staycoolcall911) [inline tsoa/sim.w:31]: Just curious (not sure it matters):
Why put it into a new bucket? Why not use the `sim.State`?

File: tsoa/app.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
+const { Socket } = require("net");
 const express = require("express");
 const { setLifted } = require("./clients");
 
-exports.runServer = async (routes, clients) => {
+exports.runServer = async (routes, clients, lastPort) => {
```
Comment (staycoolcall911) [inline tsoa/app.js:5]: Can we add a test for that?

**Review by staycoolcall911 — APPROVED** [review]
LGTM!

**Pull request merged**